extensions/amp-video/0.1/video-cache.js

```diff
@@ -85,7 +88,27 @@ function selectVideoSource(videoEl) {
  */
 function applySourcesToVideo(videoEl, sources, maxBitrate) {
   sources
-    .sort((a, b) => a['bitrate_kbps'] - b['bitrate_kbps'])
+    .sort((a, b) => {
+      // 'codec' values can contain metadata after the '.' that we must strip
+      // for sorting purposes. For example, "vp09.00.30.08" contains level,
+      // profile, and color depth values that are ignored in this sort.
+      const aCodec = a['codec']?.split('.')[0];
+      const bCodec = b['codec']?.split('.')[0];
+
+      for (let i = 0; i < CODECS_IN_DESCENDING_PRIORITY.length; i++) {
+        const codec = CODECS_IN_DESCENDING_PRIORITY[i];
+        if (aCodec === codec && bCodec !== codec) {
+          // A positive value results in a being sorted before b.
+          return 1;
+        } else if (aCodec !== codec && bCodec === codec) {
+          // A negative value results in b being sorted before a.
+          return -1;
+        }
+      }
+
+      // If neither codec has greater priority, sort by descending bitrate.
+      return a['bitrate_kbps'] - b['bitrate_kbps'];
```

Comment:
Most of the time codecs will be equal, would it make sense to run this line before the `for` loop inside an extra condition, and with an early return? (cf prev comment)